stackslib/src/chainstate/stacks/db/blocks.rs

```diff
@@ -4150,25 +4152,89 @@ impl StacksChainState {
                 block_height,
                 txid,
                 burn_header_hash,
+                signer_key,
+                max_amount,
+                auth_id,
                 ..
             } = &stack_stx_op;
+
+            let mut args = vec![
+                Value::UInt(*stacked_ustx),
+                // this .expect() should be unreachable since we coerce the hash mode when
+                // we parse the StackStxOp from a burnchain transaction
+                reward_addr
+                    .as_clarity_tuple()
+                    .expect("FATAL: stack-stx operation has no hash mode")
+                    .into(),
+                Value::UInt(u128::from(*block_height)),
+                Value::UInt(u128::from(*num_cycles)),
+            ];
+            // Appending additional signer related arguments for pox-4
+            if POX_4_NAME == active_pox_contract {
```

Comment:
Can this `if`-body be put into its own method?

Reply:
done 👍